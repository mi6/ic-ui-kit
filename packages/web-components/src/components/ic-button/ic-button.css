@import "../../global/normalize.css";

/**
* @prop --min-width: Minimum width of the button.
*/

:host {
  display: inline-block;

  --button-default: var(--ic-action-default);
  --button-default-hover: var(--ic-action-default-hover);
  --button-default-active: var(--ic-action-default-active);
  --button-default-background-hover: var(--ic-action-default-bg-hover);
  --button-default-background-active: var(--ic-action-default-bg-active);
  --icon-width: 100%;
  --icon-height: 100%;
}

.button {
  font-family: var(--ic-font-body-family);
  text-decoration: none;
  font-weight: 600;
  font-size: 0.875rem;
  transition: var(--ic-easing-transition-fast);
  border-radius: var(--ic-border-radius);
  min-width: var(--min-width, 6.25rem);
  display: inline-flex;
  gap: var(--ic-space-xs);
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background: none;
  border: none;
  box-sizing: border-box;
  white-space: nowrap;
  vertical-align: middle;
}

.button:hover {
  cursor: pointer;
}

.button:focus,
:host .button:focus {
  box-shadow: var(--ic-border-focus);
}

:host(.light) .button:focus {
  box-shadow: var(--ic-border-focus);
}

.button:focus-visible {
  outline: var(--ic-hc-focus-outline);
}

:host(.dark) .button {
  --button-default: var(--ic-action-dark);
  --button-default-hover: var(--ic-action-dark);
  --button-default-active: var(--ic-action-dark);
  --button-default-background-hover: var(--ic-action-dark-bg-hover);
  --button-default-background-active: var(--ic-action-dark-bg-active);
}

:host(.light) .button {
  --button-default: var(--ic-action-light);
  --button-default-hover: var(--ic-action-light);
  --button-default-active: var(--ic-action-light);
  --button-default-background-hover: var(--ic-action-light-bg-hover);
  --button-default-background-active: var(--ic-action-light-bg-active);
}

:host(.button-variant-primary.light) .button {
  color: var(--ic-color-primary-text);

  --button-default-hover: var(--ic-action-light-hover);
  --button-default-active: var(--ic-action-light-active);
}

:host(.button-variant-primary.dark) .button {
  --button-default-hover: var(--ic-action-dark-hover);
  --button-default-active: var(--ic-action-dark-active);
}

:host(.disabled),
:host(.disabled) .button,
:host(.loading),
:host(.loading) .button {
  pointer-events: none;
}

:host(.loading-with-icon) .button {
  min-width: var(--min-width, 8.25rem);
}

/* Variants */

/* Primary */

:host(.button-variant-primary) .button {
  color: var(--ic-architectural-white);
  background-color: var(--button-default);
}

:host(.button-variant-primary) .button:hover:not(:focus) {
  background-color: var(--button-default-hover);
}

:host(.button-variant-primary) .button:active:not(:focus),
:host(.button-variant-primary.loading) .button {
  background-color: var(--button-default-active);
}

:host(.button-variant-primary.disabled) .button {
  background: var(--ic-architectural-200);
  color: var(--ic-architectural-300);
}

:host(.button-variant-primary.light.disabled) .button {
  background: var(--ic-architectural-600);
  color: var(--ic-architectural-500);
}

/* Secondary */

:host(.button-variant-secondary) .button {
  border: var(--ic-space-1px) solid var(--button-default);
  color: var(--button-default);
}

:host(.button-variant-secondary) .button:hover:not(:focus) {
  background-color: var(--button-default-background-hover);
  border-color: var(--button-default-hover);
  color: var(--button-default-hover);
}

:host(.button-variant-secondary) .button:active:not(:focus) {
  border-color: var(--button-default-active);
  background-color: var(--button-default-background-active);
  color: var(--button-default-active);
}

:host(.button-variant-secondary.loading) .button {
  border-color: var(--button-default-active);
  background-color: var(--button-default-background-active);
  color: var(--button-default-active);
}

:host(.button-variant-secondary.disabled) .button,
:host(.button-variant-secondary.disabled) .button:hover,
:host(.button-variant-secondary.disabled) .button:active {
  border-color: var(--ic-architectural-300);
  color: var(--ic-architectural-300);
  background: none;
}

:host(.button-variant-secondary.light.disabled) .button,
:host(.button-variant-secondary.light.disabled) .button:hover,
:host(.button-variant-secondary.light.disabled) .button:active {
  border-color: var(--ic-architectural-500);
  color: var(--ic-architectural-500);
  background: none;
}

/* Tertiary */

:host(.button-variant-tertiary) .button {
  color: var(--button-default);
}

:host(.button-variant-tertiary) .button:hover:not(:focus) {
  background-color: var(--button-default-background-hover);
  color: var(--button-default-hover);
}

:host(.button-variant-tertiary) .button:active:not(:focus),
:host(.button-variant-tertiary.loading) .button {
  background-color: var(--button-default-background-active);
  color: var(--button-default-active);
}

:host(.button-variant-tertiary.disabled) .button,
:host(.button-variant-tertiary.disabled) .button:hover,
:host(.button-variant-tertiary.disabled) .button:active {
  border-color: var(--ic-architectural-300);
  color: var(--ic-architectural-300);
  background: none;
}

:host(.button-variant-tertiary.light.disabled) .button,
:host(.button-variant-tertiary.light.disabled) .button:hover,
:host(.button-variant-tertiary.light.disabled) .button:active {
  border-color: var(--ic-architectural-500);
  color: var(--ic-architectural-500);
  background: none;
}

/* Icon */

:host(.button-variant-icon) .button {
  color: var(--button-default);
  min-width: 0;
}

:host(.button-variant-icon) .button:hover:not(:focus) {
  background-color: var(--button-default-background-hover);
  color: var(--button-default-hover);
}

:host(.button-variant-icon) .button:active:not(:focus),
:host(.button-variant-icon.loading) .button {
  background-color: var(--button-default-background-active);
  color: var(--button-default-active);
}

:host(.button-variant-icon.disabled) .button,
:host(.button-variant-icon.disabled) .button:hover,
:host(.button-variant-icon.disabled) .button:active {
  color: var(--ic-architectural-300);
  background: none;
}

/* Destructive */

:host(.button-variant-destructive) .button {
  color: var(--ic-architectural-white);
  background-color: var(--ic-action-destructive);
  text-transform: uppercase;
}

:host(.button-variant-destructive) .button:hover:not(:focus) {
  background-color: var(--ic-action-destructive-hover);
}

:host(.button-variant-destructive) .button:active:not(:focus),
:host(.button-variant-destructive.loading) .button {
  background-color: var(--ic-action-destructive-active);
}

:host(.button-variant-destructive.disabled) .button {
  background: var(--ic-architectural-200);
  color: var(--ic-architectural-300);
}

/* Sizing */

:host(.button-size-default) .button {
  height: 2.5rem;
  padding: var(--ic-space-xs) var(--ic-space-md);
}

:host(.button-size-small) .button {
  height: var(--ic-space-xl);
  padding: var(--ic-space-xxs) var(--ic-space-md);
}

:host(.button-size-large) .button {
  height: var(--ic-space-xxl);
  padding: var(--ic-space-sm) var(--ic-space-md);
}

:host(.button-size-default.button-variant-icon) .button {
  height: var(--ic-space-xl);
  width: var(--ic-space-xl);
  padding: 0.375rem;
}

:host(.button-size-small.button-variant-icon) .button {
  height: var(--ic-space-lg);
  width: var(--ic-space-lg);
  padding: var(--ic-space-xxs);
}

:host(.button-size-large.button-variant-icon) .button {
  height: 2.5rem;
  width: 2.5rem;
  padding: var(--ic-space-xs);
}

/* Width */
:host(.full-width),
:host(.full-width) .button {
  width: 100%;
}

/* Loading */

div.loading-container {
  position: relative;
  align-items: center;
  width: 100%;
}

ic-loading-indicator {
  --outer-color: transparent;
}

@keyframes loading-animation {
  0% {
    width: 0%;
    left: 0%;
  }

  25% {
    width: 0%;
    left: 0%;
    opacity: 0;
  }

  50% {
    width: 100%;
    left: 0%;
    opacity: 1;
  }

  75% {
    width: 0%;
    left: 100%;
  }

  100% {
    width: 0%;
    left: 100%;
    opacity: 0;
  }
}

/* Icons */

div.icon-container {
  box-sizing: border-box;
  width: var(--ic-space-lg);
  height: var(--ic-space-lg);
  display: flex;
  justify-content: center;
  align-items: center;
}

::slotted(*) {
  width: var(--icon-width) !important;
  height: var(--icon-height) !important;
  fill: currentcolor !important;
  pointer-events: none;
}

:host(.button-variant-icon) .button .icon-container {
  margin: 0;
  pointer-events: none;
}

/** SEARCH **/

:host(.search-submit-button) ::slotted(svg) {
  --icon-height: 1.25rem;
  --icon-width: 1.25rem;
}

:host(.search-submit-button-small) ::slotted(svg) {
  --icon-height: 1rem;
  --icon-width: 1rem;
}

:host(.clear-button) {
  margin: 0 var(--ic-space-xxs);
}

:host(.clear-button) .button:focus {
  box-shadow: none;
}

:host(.search-submit-button) .button:focus {
  box-shadow: none;
}

:host(.search-submit-button) .button:not(:active):focus {
  box-shadow: none;
  background-color: var(--ic-action-default-bg-hover);
}

:host(.search-submit-button) {
  display: flex;
  align-items: center;
  margin: 0 var(--ic-space-xxs);
}

:host(.menu-close-button) ::slotted(svg) {
  --icon-height: 0.875rem;
  --icon-width: 0.875rem;
}

:host(.popout-menu-button) .button {
  justify-content: left;
  border-radius: 0;
  white-space: pre-line;
  text-align: start;
}

:host(.popout-menu-button) div.icon-container {
  flex: none;
}

:host(.popout-menu-button) .button:focus {
  box-shadow: var(--ic-border-focus-inset);
  border-radius: var(--ic-border-radius-inset);
}

.ic-button-describedby {
  display: none;
}

:host .ic-tooltip {
  display: block;
}

/** High Contrast **/
@media (forced-colors: active) {
  .button {
    border: 0.125rem solid transparent;
  }
}

/** Pagination **/

:host(.flip) ::slotted(svg) {
  transform: scaleX(-1);
}
