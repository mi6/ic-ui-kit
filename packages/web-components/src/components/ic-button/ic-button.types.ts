export type IcButtonTypes = "reset" | "submit" | "button";

export type IcButtonVariants =
  | "primary"
  | "secondary"
  | "tertiary"
  | "icon"
  | "destructive";

export type IcButtonTooltipPlacement = "top" | "right" | "bottom" | "left";
