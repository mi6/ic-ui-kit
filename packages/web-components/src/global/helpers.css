html {
  font-family: var(--ic-font-body-family);
  color: var(--ic-color-text-primary);
}

a.ic-card-wrapper-link {
  text-decoration: none;
  outline: none;
  color: currentcolor;
}
