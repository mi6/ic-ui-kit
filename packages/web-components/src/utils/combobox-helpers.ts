import { IcMenuOption } from "./types";

const getOptionIdFromValue = (listboxId: string, value: string) => {
        return `${listboxId}-${value}`;
    };

export const getAriaActivedescendant = (listboxId: string, el: HTMLIcSelectNewElement | HTMLIcComboboxElement, activedescendantIndex: number | null): string => {
    const value = getValueFromActivedescendantIndex(el, listboxId, activedescendantIndex);
    return value
        ? getOptionIdFromValue(listboxId, value)
        : ""
};

export const getValueFromActivedescendantIndex = (el: HTMLIcSelectNewElement | HTMLIcComboboxElement, listboxId: string, activedescendantIndex: number | null) => {
    const descendants: (HTMLLIElement | HTMLIcButtonElement)[] = Array.from(el.shadowRoot?.querySelectorAll('[role="option"]') || []);
    const selectAllBtn = el.shadowRoot?.querySelector<HTMLIcButtonElement>(`#${getOptionIdFromValue(listboxId, "select-all")}`);
    if (selectAllBtn) descendants.unshift(selectAllBtn);

    return activedescendantIndex !== null
    ? descendants[activedescendantIndex]?.dataset.value || null
    : null
};

export const getLabelsFromValues = (valueArr: string[], flattenedOptions: IcMenuOption[]): string => {
    return valueArr.map(value => {
        return flattenedOptions.find(option => option.value === value)?.label
    }).join(", ");
};

/**
 * 
 * @returns the index of the first option that is in valueArr, otherwise 0
 */
export const getFirstSelectedOptionIndex = (flattenedOptions: IcMenuOption[], valueArr: string[]) => {
    const firstSelectedOption = flattenedOptions.find(option => {
        return valueArr.includes(option.value)
    });
    return firstSelectedOption ? flattenedOptions.indexOf(firstSelectedOption) : 0;
};

export const getDeduplicatedOptions = (options: IcMenuOption[], comboboxLabel: string): IcMenuOption[] => {
    const uniqueValues: string[] = [];
    const dedupedOptions: IcMenuOption[] = [];

    options.forEach((option) => {
        if (!option.disabled) {
            if (option.children) {
                //If an option has children, we will loop through them
                const dedupedChildren: IcMenuOption[] = [];
                option.children.forEach((child) => {
                    if (uniqueValues.includes(child.value)) {
                    console.warn(
                        `ic-select with label ${comboboxLabel} was populated with duplicate option (value: ${child.value}) which has been removed.`
                    );
                    } else {
                    uniqueValues.push(child.value);
                    dedupedChildren.push(child);
                    }
                });
                // construct a modified option, inserting the deduplicated children alongside the original information
                dedupedOptions.push({
                    ...option,
                    children: dedupedChildren,
                });
            } else {
                // If an option does not have children, assess to see if it's value has been included already
                if (uniqueValues.includes(option.value)) {
                    console.warn(
                    `ic-select with label ${comboboxLabel} was populated with duplicate option (value: ${option.value}) which has been removed.`
                    );
                } else {
                    uniqueValues.push(option.value);
                    dedupedOptions.push(option);
                }
            }
        }
    });

    return dedupedOptions;
};

export const getSortedOptions = (options: IcMenuOption[]): IcMenuOption[] => {
    let sorted: IcMenuOption[] = [];
    if (options.sort) {
        sorted = options.sort((optionA, optionB) =>
            optionA.recommended && !optionB.recommended ? -1 : 0
        );
    }
    return sorted;
};

export const getFlattenedOptions = (groupedOptions: IcMenuOption[]) => {
    const flattenedOptions: IcMenuOption[] = [];

    if (groupedOptions.length > 0) {
        groupedOptions.map((option) => {
            if (option.children) {
                option.children.map(
                    (option) => flattenedOptions.push(option)
                );
            } else {
                flattenedOptions.push(option);
            }
        });
    };
    return flattenedOptions;
};