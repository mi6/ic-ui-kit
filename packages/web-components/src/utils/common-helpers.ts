import purifier from "dompurify";
import { IC_ALL_COMPONENTS } from "./constants";

export const sanitizeHTMLString = (html: string): string => {
  return purifier.sanitize(html, {
    ADD_TAGS: IC_ALL_COMPONENTS,
  }) as unknown as string;
};
