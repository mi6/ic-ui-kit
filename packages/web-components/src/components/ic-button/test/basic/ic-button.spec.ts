import { Button } from "../../ic-button";
import { newSpecPage } from "@stencil/core/testing";
import * as helpers from "../../../../utils/helpers";

beforeAll(() => {
  jest.spyOn(console, "warn").mockImplementation(jest.fn());
});

//mocked as getThemeFromContext is always default when run in test context
const mockGetThemeFromContext = () => {
  const func = jest.fn(() => {
    return "light";
  });

  Object.defineProperty(helpers, "getThemeFromContext", {
    value: func,
  });
};

describe("button component", () => {
  it("should correctly render with text", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: "<ic-button>Button</ic-button>",
      supportsShadowDom: true,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should correctly render with custom text", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: "<ic-button>Click Me</ic-button>",
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should render correct HTML for secondary variant", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: "<ic-button variant='secondary'>Button</ic-button>",
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should render correct HTML for tertiary variant", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: "<ic-button variant='tertiary'>Button</ic-button>",
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should render correct HTML for destructive variant", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: "<ic-button variant='destructive'>Button</ic-button>",
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should render correct HTML with Icon", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `
      <ic-button>
        <svg slot='icon' xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="#000000">
        <path d="M0 0h24v24H0V0z" fill="none"/>
        <path d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"/>
        </svg>
        Button
      </ic-button>
      `,
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should render correct HTML when loading", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `
      <ic-button loading>Button</ic-button>
      `,
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should render correct HTML when loading - secondary", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `
      <ic-button loading variant="secondary">Button</ic-button>
      `,
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should render correct HTML when loading - tertiary", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `
      <ic-button loading variant="tertiary">Button</ic-button>
      `,
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should render correct HTML when loading - destructive", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `
      <ic-button loading variant="destructive">Button</ic-button>
      `,
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should render correct HTML when loading - appearance light", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `
      <ic-button loading appearance="light">Button</ic-button>
      `,
    });
    expect(page.root).toMatchSnapshot();
  });

  it("render with correct button type when defined", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `
      <ic-button type='reset'>Button</ic-button>
      `,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should render with "a" tag when href is passed', async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `
      <ic-button href = '#'>Button</ic-button>
      `,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should render with defined "a" tag props', async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `
      <ic-button href = '#' download rel="nofollow" target="_blank">Button</ic-button>
      `,
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should render icon variant with a tooltip", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: "<ic-button variant='icon' id='test-button'>Button</ic-button>",
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should render icon variant with a tooltip based on title", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: "<ic-button variant='icon' id='test-button' title='Tooltip text'>Button</ic-button>",
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should render icon variant with a tooltip based on aria-label", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: "<ic-button variant='icon' id='test-button' aria-label='Tooltip text' tooltip-placement='top'>Button</ic-button>",
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should disable tooltip when prop set", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: "<ic-button variant='icon' aria-label='Tooltip text' id='test-button' disable-tooltip>Button</ic-button>",
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should update when method called", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: "<ic-button variant='icon' aria-label='Tooltip text' id='test-button'>Button</ic-button>",
    });

    expect(page.root).toMatchSnapshot();

    await page.root.updateAriaLabel("New tooltip text");
    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  it("should stop immediate propagation of a click event when disabled", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: "<ic-button id='test-button' disabled=true onclick='alert('test')'>Button</ic-button>",
    });

    jest.spyOn(window, "alert").mockImplementation();

    const element = document.getElementById("test-button");

    element.click();

    await page.waitForChanges();

    expect(window.alert).not.toHaveBeenCalled;
  });

  it("should call 'setFocus' when button is focussed", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: "<ic-button id='ic-button'>Button</ic-button>",
    });

    //Can't expect anything in this test - this is to increase code coverage only
    await page.rootInstance.setFocus().toHaveBeenCalled;
  });

  it("should test theme change", async () => {
    mockGetThemeFromContext();

    const page = await newSpecPage({
      components: [Button],
      html: `<ic-button id='ic-button'>Button</ic-button>`,
    });

    await page.rootInstance.themeChangeHandler({
      detail: { mode: "light" },
    });
    await page.waitForChanges();

    expect(page.rootInstance.appearance).toEqual("light");
  });

  it("should test button as submit button on form", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `<form id="new-form"><form><ic-button id='ic-button' type="submit" form="new-form">Button</ic-button>`,
    });

    const btn = document.getElementById("ic-button");
    btn.click();

    expect(page.root).toMatchSnapshot();
  });

  it("should test aria-describedby is set", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `<span id="button-description">
        This is the button description
      </span>
      <ic-button id='ic-button' aria-describedby="button-description">Button</ic-button>
      `,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should test aria-describedby is updated", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `<span id="button-description">
        This is the button description
      </span>
      <ic-button id='ic-button' aria-describedby="button-description">Button</ic-button>
      `,
    });

    (document.querySelector("#button-description") as HTMLElement).innerText =
      "This is the new description";
    page.rootInstance.mutationCallback();

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  it("should test blur handler", async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `<ic-button id='ic-button'>Button</ic-button>`,
    });

    const eventSpy = jest.fn();
    page.win.addEventListener("icBlur", eventSpy);

    await page.rootInstance.onBlur();
    await page.waitForChanges();
    expect(eventSpy).toHaveBeenCalled();
  });
});
