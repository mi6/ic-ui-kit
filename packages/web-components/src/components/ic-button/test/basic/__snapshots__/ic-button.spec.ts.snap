// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component render with correct button type when defined 1`] = `
<ic-button class="button-size-default button-variant-primary" exportparts="button" type="reset">
  <mock:shadow-root>
    <button class="button" part="button" type="reset">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should correctly render with custom text 1`] = `
<ic-button class="button-size-default button-variant-primary" exportparts="button">
  <mock:shadow-root>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Click Me
</ic-button>
`;

exports[`button component should correctly render with text 1`] = `
<ic-button class="button-size-default button-variant-primary" exportparts="button">
  <mock:shadow-root>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should disable tooltip when prop set 1`] = `
<ic-button class="button-size-default button-variant-icon" disable-tooltip="" exportparts="button" id="test-button" variant="icon">
  <mock:shadow-root>
    <button aria-label="Tooltip text" class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should render correct HTML for destructive variant 1`] = `
<ic-button class="button-size-default button-variant-destructive" exportparts="button" variant="destructive">
  <mock:shadow-root>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should render correct HTML for secondary variant 1`] = `
<ic-button class="button-size-default button-variant-secondary" exportparts="button" variant="secondary">
  <mock:shadow-root>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should render correct HTML for tertiary variant 1`] = `
<ic-button class="button-size-default button-variant-tertiary" exportparts="button" variant="tertiary">
  <mock:shadow-root>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should render correct HTML when loading - appearance light 1`] = `
<ic-button appearance="light" class="button-size-default button-variant-primary light loading" exportparts="button" loading="">
  <mock:shadow-root>
    <button aria-disabled="true" aria-label="Loading" class="button" part="button" type="button">
      <div class="loading-container">
        <ic-loading-indicator appearance="light" type="linear"></ic-loading-indicator>
      </div>
    </button>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should render correct HTML when loading - destructive 1`] = `
<ic-button class="button-size-default button-variant-destructive loading" exportparts="button" loading="" variant="destructive">
  <mock:shadow-root>
    <button aria-disabled="true" aria-label="Loading" class="button" part="button" type="button">
      <div class="loading-container">
        <ic-loading-indicator appearance="light" type="linear"></ic-loading-indicator>
      </div>
    </button>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should render correct HTML when loading - secondary 1`] = `
<ic-button class="button-size-default button-variant-secondary loading" exportparts="button" loading="" variant="secondary">
  <mock:shadow-root>
    <button aria-disabled="true" aria-label="Loading" class="button" part="button" type="button">
      <div class="loading-container">
        <ic-loading-indicator appearance="dark" type="linear"></ic-loading-indicator>
      </div>
    </button>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should render correct HTML when loading - tertiary 1`] = `
<ic-button class="button-size-default button-variant-tertiary loading" exportparts="button" loading="" variant="tertiary">
  <mock:shadow-root>
    <button aria-disabled="true" aria-label="Loading" class="button" part="button" type="button">
      <div class="loading-container">
        <ic-loading-indicator appearance="dark" type="linear"></ic-loading-indicator>
      </div>
    </button>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should render correct HTML when loading 1`] = `
<ic-button class="button-size-default button-variant-primary loading" exportparts="button" loading="">
  <mock:shadow-root>
    <button aria-disabled="true" aria-label="Loading" class="button" part="button" type="button">
      <div class="loading-container">
        <ic-loading-indicator appearance="light" type="linear"></ic-loading-indicator>
      </div>
    </button>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should render correct HTML with Icon 1`] = `
<ic-button class="button-size-default button-variant-primary" exportparts="button">
  <mock:shadow-root>
    <button class="button" part="button" type="button">
      <div class="icon-container">
        <slot name="icon"></slot>
      </div>
      <slot></slot>
    </button>
  </mock:shadow-root>
  <svg fill="#000000" height="24px" slot="icon" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
    <path d="M0 0h24v24H0V0z" fill="none"></path>
    <path d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"></path>
  </svg>
  Button
</ic-button>
`;

exports[`button component should render icon variant with a tooltip 1`] = `
<ic-button class="button-size-default button-variant-icon" exportparts="button" id="test-button" variant="icon">
  <mock:shadow-root>
    <ic-tooltip placement="bottom" target="ic-button-with-tooltip-test-button">
      <button aria-describedby="ic-tooltip-ic-button-with-tooltip-test-button" class="button" id="ic-button-with-tooltip-test-button" part="button" type="button">
        <slot></slot>
      </button>
    </ic-tooltip>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should render icon variant with a tooltip based on aria-label 1`] = `
<ic-button class="button-size-default button-variant-icon" exportparts="button" id="test-button" tooltip-placement="top" variant="icon">
  <mock:shadow-root>
    <ic-tooltip label="Tooltip text" placement="top" target="ic-button-with-tooltip-test-button">
      <button aria-describedby="ic-tooltip-ic-button-with-tooltip-test-button" aria-label="Tooltip text" class="button" id="ic-button-with-tooltip-test-button" part="button" type="button">
        <slot></slot>
      </button>
    </ic-tooltip>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should render icon variant with a tooltip based on title 1`] = `
<ic-button class="button-size-default button-variant-icon" exportparts="button" id="test-button" variant="icon">
  <mock:shadow-root>
    <ic-tooltip label="Tooltip text" placement="bottom" target="ic-button-with-tooltip-test-button">
      <button aria-describedby="ic-tooltip-ic-button-with-tooltip-test-button" class="button" id="ic-button-with-tooltip-test-button" part="button" type="button">
        <slot></slot>
      </button>
    </ic-tooltip>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should render with "a" tag when href is passed 1`] = `
<ic-button class="button-size-default button-variant-primary" exportparts="button" href="#">
  <mock:shadow-root>
    <a class="button" href="#" part="button">
      <slot></slot>
    </a>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should render with defined "a" tag props 1`] = `
<ic-button class="button-size-default button-variant-primary" download="" exportparts="button" href="#" rel="nofollow" target="_blank">
  <mock:shadow-root>
    <a class="button" download="" href="#" part="button" rel="nofollow" target="_blank">
      <slot></slot>
    </a>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should test aria-describedby is set 1`] = `
<ic-button class="button-size-default button-variant-primary light" exportparts="button" id="ic-button">
  <mock:shadow-root>
    <button aria-describedby="button-description" class="button" part="button" type="button">
      <slot></slot>
    </button>
    <span class="ic-button-describedby" id="button-description">
      This is the button description
    </span>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should test aria-describedby is updated 1`] = `
<ic-button class="button-size-default button-variant-primary light" exportparts="button" id="ic-button">
  <mock:shadow-root>
    <button aria-describedby="button-description" class="button" part="button" type="button">
      <slot></slot>
    </button>
    <span class="ic-button-describedby" id="button-description">
      This is the new description
    </span>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should test button as submit button on form 1`] = `
<ic-button class="button-size-default button-variant-primary light" exportparts="button" form="new-form" id="ic-button" type="submit">
  <mock:shadow-root>
    <button class="button" form="new-form" part="button" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should update when method called 1`] = `
<ic-button class="button-size-default button-variant-icon" exportparts="button" id="test-button" variant="icon">
  <mock:shadow-root>
    <ic-tooltip label="Tooltip text" placement="bottom" target="ic-button-with-tooltip-test-button">
      <button aria-describedby="ic-tooltip-ic-button-with-tooltip-test-button" aria-label="Tooltip text" class="button" id="ic-button-with-tooltip-test-button" part="button" type="button">
        <slot></slot>
      </button>
    </ic-tooltip>
  </mock:shadow-root>
  Button
</ic-button>
`;

exports[`button component should update when method called 2`] = `
<ic-button class="button-size-default button-variant-icon" exportparts="button" id="test-button" variant="icon">
  <mock:shadow-root>
    <ic-tooltip label="Tooltip text" placement="bottom" target="ic-button-with-tooltip-test-button">
      <button aria-describedby="ic-tooltip-ic-button-with-tooltip-test-button" aria-label="New tooltip text" class="button" id="ic-button-with-tooltip-test-button" part="button" type="button">
        <slot></slot>
      </button>
    </ic-tooltip>
  </mock:shadow-root>
  Button
</ic-button>
`;
